src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -88,7 +88,7 @@ public static class AttributeMatcher
             {
                 if (!context.AllAttributes.ContainsKey(attribute) ||
                     context.AllAttributes[attribute] == null ||
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute] as string))
+                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute].ToString()))
```

Comment:
don't have a `bool` _HTML_ attribute but do have a bound attribute that cannot be checked here because it is not `Nullable<T>`.  must either make the property `bool?` or treat the corresponding mode as having no required attributes.  @DamianEdwards ?